### ISSUE TITLE
Remove flex/bison parser

### ISSUE THREAD

**Issue by bchurchill** [StanfordPL/x64asm#76]
In the parser, the to_instr() function has been taking operands along with their x64asm::Type, passed in as an array of pairs.  We can eliminate this redundancy once #70 is done, and also all the typed operand productions in the grammar.

**Comment by eschkufz** [comment]
For this matter, we're not all that far from being able to pitch the whole flex/bison parser altogether now.

I'd say that at this point it's better to just replace this issue with one to eliminate the legacy code. This issue looks low priority enough (no one has touched it in 3 months) that I think it's okay.

**Comment by bchurchill** [comment]
how would we parse assembly code without the flex/bison parser?  I see no reason to throw it out?

**Comment by eschkufz** [comment]
Just with little hand-written c-codes.

I was just making the observation that a lot of the code in the library was
tending towards this dispatch on type structure -- see the register print
code -- and you could just as soon write the corresponding dispatch on type
register read code for example.

There are some nasty artifacts associated with using a flex/bison parser in
a larger c++ project. Manolis and I came upon some weird segfault that had
to do with static initialization that neither of us could diagnose. It
never turned into an issue, but I could imagine something like that
happening in the future.

It's just one less dependency to support, but probably not worth worrying
about unless there was nothing else to worry about first.

**Comment by bchurchill** [comment]
Gotcha.  I might be overhauling the parser soon anyway so I could do this.
